### DISCUSSION TITLE
How to prevent HTML in markdown Body on Alias

### DISCUSSION THREAD

**Discussion by Syed-Hasnain-Askari** [OrchardCMS/OrchardCore#12280] [Q&A]
Folks, I have content Type in which markdown Body part also markdown is add, for example the Alias of that content type is /software/register. When I type "~/register" instead of /article/register so all the HTML shows in browser window, see below:

![image](https://user-images.githubusercontent.com/47834896/186919726-0295eeaa-c7cd-4182-bf94-e8f60fdebb69.png)

but when we go with proper alias so page render proper design where we have html from also:

![image](https://user-images.githubusercontent.com/47834896/186951476-ffaef477-7553-4953-b397-57750b4b94e3.png)

I want to prevent to show the raw html please advise @ns8482e @hishamco @Skrypt @sebastienros 
Need your kind attention on above.

**Comment by Skrypt** [comment]
Have you tried with `@Html.Raw(yourMarkdownFieldName);`
Else with Liquid it is also documented

https://docs.orchardcore.net/en/dev/docs/reference/modules/Liquid/#html-escaping
https://docs.orchardcore.net/en/dev/docs/reference/modules/Liquid/#shortcode

**Reply by saleemhaq** [reply]
> Have you tried with `@Html.Raw(yourMarkdownFieldName);`

yes we used that currently: @Html.Raw(yourMarkdownFieldName); and used the razor syntax razor pages not used the liquid:

current page url is:
https://localhost:44300/article/register
but user when go with below url:
"https://localhost:44300/register" instead of https://localhost:44300/article/register so markdown is show so how to prevent please advise @Skrypt and folks

**Reply by saleemhaq** [reply]
> Have you tried with `@Html.Raw(yourMarkdownFieldName);` Else with Liquid it is also documented
> 
> https://docs.orchardcore.net/en/dev/docs/reference/modules/Liquid/#html-escaping https://docs.orchardcore.net/en/dev/docs/reference/modules/Liquid/#shortcode

@Skrypt @ns8482e please advise on above comments.

**Reply by ns8482e** [reply]
> instead of https://localhost:44300/article/register so markdown is show

do you have any other content item that has `register `as alias?  Ideally it should return 404

can you try changing the alias to single path meaning instead of article/register change to article-register

**Reply by saleemhaq** [reply]
> > instead of https://localhost:44300/article/register so markdown is show
> 
> do you have any other content item that has `register `as alias? Ideally it should return 404
> 
> can you try changing the alias to single path meaning instead of article/register change to article-register

@ns8482e I set the preview setting in that way below:

![image](https://user-images.githubusercontent.com/5202291/187508152-0f7e37f2-726e-4f42-afcd-ab091c76db89.png)

And on the Razor page level I set it i that way below:

![image](https://user-images.githubusercontent.com/5202291/187508345-0780677a-3ed1-40ff-b078-004e0d31f474.png)

So when any new form I create inside article so make their alias like : below:

For register:
http://localhost:44300/article/Register

For Contact us:

http://localhost:44300/article/contactus

When skip the /article in whole url ie: http://localhost:44300/contactus

so it occur the below issue:

![image](https://user-images.githubusercontent.com/5202291/187509044-51e3240d-87a6-47ee-84c8-3ea93c12d133.png)

and when go with fully qualified url i.e: http://localhost:44300/article/contactus it showing correct behaviour, I want that it not show like above just show 404, please advise and help in that @ns8482e @sebastienros @hishamco @Skrypt

**Reply by saleemhaq** [reply]
Folks please help me above it still showing @ns8482e @jtkech @Skrypt
